examples/translate-all/src/localizer.js

```diff
@@ -0,0 +1,63 @@
+import yandexTranslator from './yandex-translator'
+import {arrayLocalizableToTextLocalizables, localeToLanguage} from './helpers.js'
+
+export default class Localizer {
+  constructor (options) {
+    this._options = options
+  }
+
+  localize (localizables) {
+    const promises = []
+    for (let localizable of localizables) {
+      const p = this._handleLocalizable(localizable)
+      promises.push(p)
+    }
+    return Promise.all(promises)
+  }
+
+  _handleLocalizable (localizable) {
+    const {srcValue, valueType} = localizable
+    if (
+      ['Symbol', 'Text'].indexOf(valueType) > -1 &&
+      typeof srcValue === 'string'
+    ) {
+      return this._translateLocale(localizable)
+    } else if (
+      valueType === 'Array' && Array.isArray(srcValue)
+    ) {
+      const textLocalizables = arrayLocalizableToTextLocalizables(localizable)
```

Comment:
I think it is the responsibility of the `LocalizableGenerator` to call this method.